lib/feature_management.rb

```diff
@@ -34,7 +34,7 @@ def self.development_and_telephony_test_adapter?
   end
 
   def self.prefill_otp_codes_allowed_in_sandbox?
-    Identity::Hostdata.domain == 'identitysandbox.gov' && telephony_test_adapter?
+    Identity::Hostdata.domain.end_with?('identitysandbox.gov') && telephony_test_adapter?
```

Comment:
```suggestion
    (Identity::Hostdata.domain == ('identitysandbox.gov')
      || Identity::Hostdata.domain.end_with? ('.identitysandbox.gov'))
      && telephony_test_adapter?
```

Maybe something like this?

Reply:
Alternatively with `RegExp`?

```suggestion
    /\.?identitysandbox\/.gov$/.match?(Identity::Hostdata.domain) && telephony_test_adapter?
```

Reply:
I think the regular expression has the same issue

```ruby
regex = /\.?identitysandbox\.gov$/
regex.match?('identitysandbox.gov')
# => true
regex.match?('aidentitysandbox.gov')
# => true
regex.match?('hacker-identitysandbox.gov')
# => true
regex.match?('tooling.identitysandbox.gov')
# => true
```

Reply:
> I think the regular expression has the same issue

Good catch. A better pattern might be `/\.?identitysandbox\.gov$/`

(Not proposing we change from current implementation)